client/data/themes/use-tier-retained-benefits-query.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@tanstack/react-query';
+import wpcom from 'calypso/lib/wp';
+
+export type RetainedBenefit = {
+	is_eligible: boolean;
+	eligibility_expiration_date: number;
+	tier: {
+		slug: string;
+		feature: string | null;
+		platform: string;
+	};
+};
+
+export type TierRetainedBenefits = {
+	[ index: string ]: {
+		retained_benefits: RetainedBenefit;
+	};
+};
+
+export const useTierRetainedBenefitsQuery = (
+	siteId: number | null,
+	themeSlug: string
+): RetainedBenefit | undefined => {
+	const queryKey = [ 'themeTierRetainedBenefits', siteId ];
```

Comment:
This could be a constant outside the function. That way we would avoid creating it every time `useTierRetainedBenefitsQuery` is called.

Reply:
We can't do that because we pass `siteId`, which is passed as an argument.

Reply:
Oh, that's right haha.